pkg/kubernetes/scripts/virtual-machines/components/VmActions.jsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// @flow
+
+import React, { PropTypes } from 'react';
+import { gettext as _ } from 'cockpit';
+import { connect } from 'react-redux'
+
+import type { Vm } from '../types.jsx';
+import { vmIdPrefx } from '../utils.jsx';
+import { vmShutdown } from '../kube-middleware.jsx';
+import { vmActionFailed } from '../action-creators.jsx';
+
+// import DropdownButtons from '../../../../machines/components/dropdownButtons.jsx';
+import { mouseClick } from '../utils.jsx';
+
+React;
+
+function canShutdown({ vm }: { vm: Vm }) {
+  vm ; // TODO: check on state
+  return true;
+}
+
+const VmActions = ({ vm, onFailure }: { vm: Vm, onFailure: Function }) => {
+  const id = vmIdPrefx(vm);
+
+  const onShutdown = () => {
+    vmShutdown({ vm }).catch((error) => {
+      console.info('VmActions: shutdown failed: ', error);
+      onFailure({
+        message: _("VM Shut Down failed."),
+        detail: error,
+      });
+    });
+  };
+
+  let shutdown = null;
+  if (canShutdown({ vm })) {
+    shutdown = (
+      <button className="btn btn-default btn-danger" onClick={mouseClick(onShutdown)} id={`${id}-shutdown`}>
+        {_("Shut Down")}
```

Comment:
Is the naming right? + rest of the code
This will delete the vm so it could be a bit confusing.

Reply:
It is correct, removal of VirtualMachine object from etcd leads to its shut down. "Long-term store of VM definitions" is implemented via OfflineVirtualMachines in kubevirt. So HTTP DELETE is ok even if semantics is `Shut Down` from user perspective.

Reply:
> So HTTP DELETE is ok even if semantics is Shut Down from user perspective.

Speaking solely of k6t v0.2: The button invokes action that not only shuts the VM down but also remotes it from the system entirely, not leaving user possibility to start the VM again. I'm worried that it may be (unpleasantly) surprising action for users for a button called "Shut Down".

Reply:
I agree with @suomiy and @jniederm ; the tests also confirm that this has "delete" semantics, not like the shutdown on the "Machines" page. So calling the  button that sounds safer for now?

Reply:
ok, terminology ... 

In kubevirt 0.2.0 the `shut down` was defined as `delete`. I agree its confusing, but its as designed - there is no concept of "long-term" VM definitions in kubevirt 0.2.0, the only  VirtualMachine objects are those running. This will be changed by `OfflineVirtualMachines`, but it's under development now.

Since semantics is about to be changed for a later kubevirt anyway, I'll reword to `delete`.